File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        L,R,N=0,0,len(nums)
        ans=0
        curS=0 
        ac=accumulate(nums)
       
        mp={0:1}
        for x in ac:
            if x-k in mp:
                ans+=mp[x-k]
            mp[x]=mp.get(x,0)+1
            
        
        return ans
                
        

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        int count = 0;

        for (int i = 0; i < nums.size(); i++) {
            long long currentSum = 0;
            for (int j = i; j < nums.size(); j++) {
                currentSum += nums[j];
                if (currentSum == k) count++;
            }
        }

        return count;
    }
};